sig-apps/Charter.md

```diff
@@ -0,0 +1,141 @@
+# SIG Apps Charter
+
+## Scope
+
+SIG Apps Covers deploying and operating applications in Kubernetes with a focus on the application developer and application operator experience.
+
+This includes but is not limited to:
+
+* Discussions on all levels from application development, testing environments, and production
+* Discussion of how to define and run apps in Kubernetes
+* Demos of relevant tools and projects
+* Discussion on areas of friction that can lead to suggesting improvements or feature requests
+* Development of Kubernetes primitives (e.g., Workloads API) to enable the application developers and operators. Note, most new types should be created as CRDs during their development phase and may always be ecosystem projects
+* Creation of documentation to enable application developers and operators
+* Development and maintenance of the sub-projects of Helm, Charts, and Kompose. Note, these projects are grandfathered and treated as part of the ecosystem but owned by SIG Apps. New projects are expected to stay in the ecosystem.
+
+### Goals
+
+* Discuss running and defining applications in Kubernetes (e.g., APIs, SDKs, Controllers, package management tools, etc.)
```

Comment:
Controllers is not all controllers, just the Replication Controller, correct?

There is a proposal to own SDKs in the potential SIG Platform-Dev. Does this conflict with their proposal? 

It might also be explicit to state that the APIs in question are `apps/v1` and not all APIs.

Reply:
SIG Apps owns all the controllers for the apps and batch api groups. Agree that we should explicitly mention the API groups SIG Apps here. We previously had discussions around third-party APIs, controllers, SDKs but more in-depth discussions around these will move to SIG Platform-Dev. There will continue to be cross-collaboration on these fronts as defined in their proposal: https://groups.google.com/forum/#!topic/kubernetes-users/uVFndwjYGgo. I'm not sure how best to word this to reflect all of that.

Reply:
The code that SIGs own is recorded elsewhere and open to change over time. I'm not sure we could couple the scope of the charter with the code and API a SIG owns. For reference, the code is considered to be a subproject and that's recorded in the [sigs.yaml](https://github.com/kubernetes/community/blob/master/sigs.yaml) file.

For example, SIG Apps has the ability to add a new subproject per processes in the charter. That can be recorded in the sigs.yaml file. The charter shouldn't need to change to add a subproject.

I defer to the @kubernetes/steering-committee on this one. cc @michelleN @jbeda

Reply:
/cc @bgrant0607 @smarterclayton

Reply:
Do you anticipate SIG Apps owning more core API groups and controllers often? An amendment to the core APIs or Controllers that SIG Apps owns has never happened, so I think that would constitute a charter amendment.

Reply:
This is a good question. I'm not entirely sure. Here are 3 things that I'm thinking about:

1. When SIG Apps started some of the Workload APIs didn't exist. APIs have been added.
1. Kubernetes core development is slowing down with fewer features going in. Someone in SIG Architecture described it heading toward boring infrastructure the same way folks have wanted to see Docker go. With innovation up the stack while Kubernetes was stable.
1. There have been talks of v2 APIs and new API groups. These are more about groups and better APIs than feature changes. Note, there are no plans for the v1 APIs to go away anytime soon.

There's a certain amount of uncertainty here. Due to that and the separation of a charter from the explicit documentation of everything owned I'd like some feedback from the steering committee.

Reply:
Its unclear what the point of the charter is other than to answer specifically these types of questions.

Reply:
> Its unclear what the purpose of the charter is other than to specifically answer these types of questions.

@r2d4 That's a good question.

When SIG Apps was formed, a couple years ago, like other SIGs it had to have a scope.

At the last contributor summit, with KubeCon in December, there were a number of conversations on governance. Questions came up such as, who is a member of a SIG? Are the processes they use documented? These are all part of a broader effort, by the steering committee, to clarify and document the kubernetes governance. 

Recently, the steering committee put a task to the SIGs to clarify these elements in a charter. The template they provided to the SIGs, which this document used as a starting point, didn't include the scope. SIGs have been adding their existing scope to the charter because scope is typically part of a charter.

From what I can tell, this effort is not around re-defining the scope of SIGs. It's an effort to document processes within a SIG.

Reply:
The governance.md document at the root of this repository clearly states the purpose of this document is to define scope.

> Each SIG must have a charter that specifies its scope (topics,
subsystems, code repos and directories), responsibilities, areas of
authority, how members and roles of authority/leadership are
selected/granted, how decisions are made, and how conflicts are
resolved.

https://github.com/kubernetes/community/blame/master/governance.md#L64-L68

Reply:
SIG Apps owns workload APIs and their controllers. "Workloads" are categories of applications, such as stateless, stateful, and batch, rather than specific applications, such as MySQL. If we need new kinds of workload APIs and controllers in the future, SIG Apps would own them. They might fit into existing API groups or require new ones. But  the bar for adding new APIs, especially built-in ones, will be fairly high. I believe the current controllers are general enough to cover a wide variety of applications.

I'm inclined to add additional subproject information, such as subproject descriptions, to sigs.yaml rather than into charters. General scope, categories of subprojects, and key example subprojects should be covered in the charter.

Reply:
governance.md needs to be reconciled with the new charter requirements and template.

Lots of work to do.

Reply:
With respect to "Discuss..."

This SIG should own the responsibility of ensuring that the most important categories of workloads run well and are easily managed on Kubernetes, while allowing a diverse variety of application topologies and deployment tools and workflows.